### PR TITLE
Add support for CMake

### LINKED ISSUE #31: [build] switch from qmake to cmake
AFAIU `qmake` is considered obsolete and should be replaced with `cmake`.

A quick research shows that there are scripts which translate the `.pro` file, so this shouldn't be that difficult:
* [pro2cmake](https://wiki.qt.io/CMake_Port/Porting_Guide)
* [qmake2cmake](https://pypi.org/project/qmake2cmake/)

if this doesn't work for the Qt5 build, then I do not see a problem to just switch the Qt6 build and have both build systems side-by-side. The maintenance effort should be low.

### PR DESCRIPTION
**Pull request by stefanb2**
- renamed `include/ui_helpers.h` to avoid conflict with `CMAKE_AUTOUIC`
- converted .pro file with [qmake2cmake](https://pypi.org/project/qmake2cmake/)
- use `CMAKE_AUTORCC` instead of the generated resources code
- remove generation & install of deploy script
- DRY removal: set `TARGET` variable and replace all hard-coded name references with it.
- refactor the generated monolithic CMakeLists.txt into one file per source code directory
- disable adding of `RPATH`
- add custom build target `test` to run tests
- updated the documentation accordingly
- fix compiler warnings
    
Fixes #31

**NOTES:**
* ~not tested with Qt5~ :heavy_check_mark: works now with Qt5 and Qt6 < 6.3.0 (I can only test with 5.15.x & 6.5.x).
* not tested under Windows. Is that build still supported?
* if above issues are resolved, should qmake support be removed instead of supporting two different build system in parallel?

### CODE REVIEW

**Comment by stefanb2** [comment]
With all patches applied on top of 1.2.4 I now get clean RPM builds for F38, F39 & rawhide, i.e. no warnings anymore.

@bkueng can you please comment on the open question if the Windows build is still supported?

**Comment by bkueng** [comment]
Cool, thanks for looking into this, much appreciated!

> not tested under Windows. Is that build still supported?

Yes I would try to still support it if not too much of a hassle. I can have a look, but I'm a bit busy currently. Adding some CI would be nice.

> if above issues are resolved, should qmake support be removed instead of supporting two different build system in parallel?

Let's just remove qmake completely.

**Comment by stefanb2** [comment]
> Yes I would try to still support it if not too much of a hassle. I can have a look, but I'm a bit busy currently. Adding some CI would be nice.

Adding GitHub actions shouldn't be a big issue. I think Windows may require travis CI, but I could be wrong there...

**Comment by stefanb2** [comment]
Not sure about translation files in AppImage. The files are included, but not used by the application. Maybe something has changed in the Qt AppImage generation process that needs to be updated in the script.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	TS_FILES ${TS_FILES}
+    )
+endif()
+
+add_subdirectory(include)
+add_subdirectory(src)
+add_subdirectory(test)
+add_subdirectory(ui)
+
+if(LINUX)
```
Comment (bkueng) [inline CMakeLists.txt:55]: This could be made compiler-specific (`if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")`)

File: src/main_class.cpp [hunk 2]
```diff
@@ -160,20 +160,25 @@ void CMain::loadTranslation() {
 	QString src_app_trans_path = bin_path + QLatin1String("/data/translations");
```
Comment (bkueng) [inline src/main_class.cpp]: I had to change this path:
```suggestion
	QString src_app_trans_path = bin_path + QLatin1String("/translations");
```

**Review by bkueng — COMMENTED** [review]
Looks very clean.
I only noticed the translation path that requires to be changed.

File: src/main_class.cpp (hunk 2 above)
Comment (stefanb2) [reply src/main_class.cpp]: Yes, makes sense.

I finally figured out why I couldn't get the translations to work. Me stupid of course assumed Qt would honor standard locales, i.e. something like `LANG=de_DE.UTF-8` would work. But no, no, no, you have to set `LANGUAGE=`. 

I verified that translations work with plain build and AppImage.

File: CMakeLists.txt (hunk 1 above)
Comment (stefanb2) [reply CMakeLists.txt:55]: Sure, but it was the same in the original `.pro` file

**Review by bkueng — APPROVED** [review]

**Pull request merged**